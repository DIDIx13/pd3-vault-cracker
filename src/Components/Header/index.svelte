<script>
    import HelpIcon from '@Components/Icons/Help.svelte';
    import GitubIcon from '@Components/Icons/Github.svelte';
    import Modal from '@Components/Modal/index.svelte';

    let showModal = false;

    function onHelpClick() {
        showModal = true;
    }
</script>

<style src="./style.scss">

</style>

<div class="background">
    <div class="wrapper">
        <div class="header">
            <div class="logo">PD3 Vault Cracker</div>
            <div class="icons">
                <div
                    class="icon pointer"
                    title="Help"
                    on:click={onHelpClick}
                    on:keydown={onHelpClick}>
                    <HelpIcon />
                </div>

                <a
                    href="https://github.com/SavageCore/pd3-vault-cracker"
                    target="_blank">
                    <div class="icon" title="View source">
                        <GitubIcon />
                    </div>
                </a>
            </div>
        </div>
    </div>
</div>

<Modal bind:showModal>
    <h2 slot="header">Help</h2>

    <p>
        This tool helps you to find the correct combination for the vault in
        Payday 3.
    </p>
    <br />
    <p>
        Enter the 2, 3 or 4 digits from the fingerprints on the vault keypad
        then click the
        <span class="green">✔</span>
        button to calculate the possible combinations.
    </p>
    <br />
    <p>
        To remove the entered digits click the
        <span class="red">✘</span>
        button.
    </p>
    <br />
    <h3>Keybinds</h3>
    <br />
    <ol>
        <li>Enter/Return - Calculate combinations</li>
        <li>Backspace/Delete - Remove last entered digit</li>
        <li>Escape - Clear entered digits</li>
    </ol>
    <br />
    <h3>Issues</h3>
    <br />
    <p>
        Please post any issues on
        <a
            href="https://github.com/SavageCore/pd3-vault-cracker/issues"
            target="_blank">
            GitHub
        </a>
        or DM me
        <a
            href="https://www.reddit.com/message/compose/?to=SavageCore"
            target="_blank">
            /u/SavageCore.
        </a>
    </p>
</Modal>
