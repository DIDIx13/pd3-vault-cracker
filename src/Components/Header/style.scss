/* style.scss */

.background {
    background-color: #1c1b22;
    width: 100%; /* Full width */
}

.header {
    font-family: "Roboto", sans-serif;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px 20px;
    color: #fff;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

.wrapper {
    max-width: 1360px; /* Set a max-width for content */
    margin: 0 auto;
    padding: 0 20px;
}

.logo {
    font-size: 24px;
    font-weight: bold;
}

.icons {
    display: flex;
    gap: 0.75rem;
}

.icon {
    width: 2rem;
    height: 2rem;
    fill: #fff;
}
