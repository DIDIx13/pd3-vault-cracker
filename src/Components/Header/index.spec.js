import Header from './index.svelte';
import { render } from '@testing-library/svelte'

describe('Header', () => {
    it('renders header', () => {
        const { container } = render(Header)

        expect(container).toMatchSnapshot()
    });
});
