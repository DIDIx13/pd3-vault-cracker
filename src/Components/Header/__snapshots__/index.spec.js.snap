// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header 1`] = `
<body>
  <div>
    <div
      class="background"
    >
      <div
        class="wrapper"
      >
        <div
          class="header"
        >
          <div
            class="logo"
          >
            PD3 Vault Cracker
          </div>
           
          <div
            class="offlinemode"
          >
            #OFFLINEMODE
          </div>
           
          <div
            class="icons"
          >
            <div
              class="icon pointer"
              title="Help"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 6a3.939 3.939 0 0 0-3.934 3.934h2C10.066 8.867 10.934 8 12
        8s1.934.867 1.934 1.934c0 .598-.481 1.032-1.216 1.626a9.208 9.208 0 0
        0-.691.599c-.998.997-1.027 2.056-1.027
        2.174V15h2l-.001-.633c.001-.016.033-.386.441-.793.15-.15.339-.3.535-.458.779-.631
        1.958-1.584 1.958-3.182A3.937 3.937 0 0 0 12 6zm-1 10h2v2h-2z"
                />
                <path
                  d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12
        2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
                />
              </svg>
            </div>
             
            <a
              href="https://github.com/SavageCore/pd3-vault-cracker"
              target="_blank"
            >
              <div
                class="icon"
                title="View source"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821
        9.465.499.09.679-.217.679-.481
        0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605
        1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084
        2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93
        0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0
        .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1
        2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099
        2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552
        4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0
        .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2
        12.026 2z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
     
    <dialog
      class="svelte-19xcnu3"
    >
      <div
        class="svelte-19xcnu3"
      >
        <h2
          slot="header"
        >
          Help
        </h2>
         
        <hr />
         
        <p>
          This tool helps you to find the correct combination for the vault in
        Payday 3.
        </p>
         
        <br />
         
        <p>
          Enter the 2, 3 or 4 digits from the fingerprints on the vault keypad
        then click the
        
          <span
            class="green"
          >
            ✔
          </span>
          
        button to calculate the possible combinations.
        </p>
         
        <br />
         
        <p>
          To remove the entered digits click the
        
          <span
            class="red"
          >
            ✘
          </span>
          
        button.
        </p>
         
        <br />
         
        <h3>
          Keybinds
        </h3>
         
        <br />
         
        <ol>
          <li>
            Enter/Return - Calculate combinations
          </li>
           
          <li>
            Backspace/Delete - Remove last entered digit
          </li>
           
          <li>
            Escape - Clear entered digits
          </li>
        </ol>
         
        <br />
         
        <h3>
          Issues
        </h3>
         
        <br />
         
        <p>
          Please post any issues on
        
          <a
            href="https://github.com/SavageCore/pd3-vault-cracker/issues"
            target="_blank"
          >
            GitHub
          </a>
          
        or DM me
        
          <a
            href="https://www.reddit.com/message/compose/?to=SavageCore"
            target="_blank"
          >
            /u/SavageCore.
          </a>
        </p>
         
        <button
          autofocus=""
          class="svelte-19xcnu3"
        >
          Close
        </button>
      </div>
    </dialog>
  </div>
</body>
`;
